code/src/App.js

```diff
@@ -1,9 +1,74 @@
-import React from 'react'
+import React, { useEffect, useState } from 'react'
+
+import { Card } from 'components/Card'
+import { MessageInput } from 'components/MessageInput'
 
 export const App = () => {
+  // App is responsible for fetching, listing, and posting the messages – as well as contains the like function (it's here because )
+  const MESSAGES_URL = "https://happy-thoughts-technigo.herokuapp.com/thoughts"
+  const [messages, setMessages] = useState([])
+
+  useEffect(() => {
+    fetchMessages()
+  }, [])
+
+  const fetchMessages = () => {
+    fetch(MESSAGES_URL)
+      .then((res) => {
+        // This is the response we're getting – but we need to transform it to JSON.
+        return res.json()
+      })
+      .then((data) => {
+        // After the response is fetched, do some filtering (we only want the "message" part of the message object to be in the array).
+        const filteredMessages = data.filter((message) => message.message)
+
+        // Limit the amount of messages visible.
+        const limitedMessages = filteredMessages.slice(0, 20)
+
+        // Call the hook function (is that what it's called?) and set 'essages' to the limited messages.
+        setMessages(limitedMessages)
+      })
+  }
+
+  const postMessage = (message) => {
+    // Fetches the messages URL, and does a post request containing the message part of the message (this is handled in the onPostMessage function which is being sent in with the MessageInput component).
+    fetch(MESSAGES_URL, {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify({ message: message })
```

Comment:
Or we can simplify this, since both key and value inside object have same name :

```
body: JSON.stringify({ message })
```